### DISCUSSION TITLE
column_filters

### DISCUSSION THREAD

**Discussion by sperrygrove** [aminalaee/sqladmin#374] [General]
Regarding "Feature parity with Flask-Admin", I've noticed that "column_filters" is omitted from this list. This feature allows a table to be filtered based on a specific field value. Related joined table fields can also be used for filtering. This could perhaps be partially achieved by implementing a search_query on the model, but wouldn't achieve searching within a pre-filtered result set.

![flask-admin-filter](https://user-images.githubusercontent.com/3480482/201095480-9bbcf2b1-f912-421d-97f3-dbf9cc6262c0.png)

**Comment by aminalaee** [comment]
Hi,
Yes you are right, this is missing.
I will try to look into it.

**Reply by sperrygrove** [reply]
Good work on the project. Looking forward to seeing how it progresses.

**Reply by ef1m0ff** [reply]
Any updates? Maybe you'll find some time to implement? This is very useful feature!

**Reply by aminalaee** [reply]
Yeah I'll try to spend some time on this one. but I can't really say when.

**Reply by abiteev-ae** [reply]
I've Spent amount of time to find documentation about this functionality, and now found this discussion, hope u will add it, cause current search is super uncomfortable for users.

**Reply by jayanthkmr** [reply]
@aminalaee Is the feature for filtering in pipeline to be picked-up?

**Comment by jonocodes** [comment]
I am also interested in this feature. In terms of design options, django admin does this on the sidebar instead of the header:

![image](https://github.com/aminalaee/sqladmin/assets/1310468/43aa6970-2d6f-4425-a362-f1a54522e36f)

and there are plugins that allow for various custom field types like this 
https://pypi.org/project/django-admin-numeric-filter/

**Comment by varvaraDev** [comment]
Hello
Just pinging to say that normal filtering is really needed, but as I understand, no one has worked on it for the past two years. :( 
Is there maybe a fork or has someone figured out a quick way to add it?

**Comment by cha0tic3ra** [comment]
Hey all, +1 to having custom_filters supported here!

**Comment by gabriel-f-santos** [comment]
Any update? let me know if need some help, i`m starting to test sqladin with fastapi

**Reply by lorg** [reply]
Hi @gabriel-f-santos , please check out my PR: https://github.com/aminalaee/sqladmin/pull/906 .

**Comment by GeorgiDimv** [comment]
Hello is there any chance to have this or maybe to contribute to implementing it

**Reply by lorg** [reply]
Hi, @GeorgiDimv please check out my PR: https://github.com/aminalaee/sqladmin/pull/906 .

**Reply by lorg** [reply]
Heya @GeorgiDimv , I saw your reply on the PR, thanks! Since I really wanted to use this feature right away, in the meantime I switched my project to depend on the branch on my forked version, hopefully this gets merged soon and becomes part of the official version.
Of course, if there would requested changes or fixes I'll be happy to make them.

**Comment by lorg** [comment]
Hi all, I created a PR for this feature here: https://github.com/aminalaee/sqladmin/pull/906
It seems that I ran into some problems with the tests and postgresql, I'll fix the issues in the next couple of days. Of course, if anyone wants to help fix this you are more than welcome :)

@aminalaee I hope you can review this and provide feedback if you want any changes. From my perspective it's already good enough to use for my own project. 

Here is an example screenshot of the feature, from a test project I set up:
<img width="1225" alt="Screenshot 2025-04-26 at 0 54 50" src="https://github.com/user-attachments/assets/b41656bd-f3f2-45bf-8316-bf0d7abd08f0" />

**Reply by lorg** [reply]
Heya, @aminalaee , the PR tests pass, would love to have it reviewed and make it into a new release :)

**Comment by GeorgiDimv** [comment]
Hey, I just tried to build with ur fork but unfortunately I cannot see the filters can you please provide a screenshot for ur working example to see for what should I look and any hints maybe I have too few records in my table in order to filter ?

Kind Regards,

From: Imri Goldberg ***@***.***>
Date: Monday, 28 April 2025 at 13:32
To: aminalaee/sqladmin ***@***.***>
Cc: Dimov, Georgi ***@***.***>, Mention ***@***.***>
Subject: Re: [aminalaee/sqladmin] column_filters (Discussion #374)

Heya @GeorgiDimv<https://github.com/GeorgiDimv> , I saw your reply on the PR, thanks! Since I really wanted to use this feature right away, in the meantime I switched my project to depend on the branch on my forked version, hopefully this gets merged soon and becomes part of the official version.
Of course, if there would requested changes or fixes I'll be happy to make them.

**Reply by lorg** [reply]
Hi @GeorgiDimv , I created a small demo project that I actually used for testing during development, I've now uploaded it here. 
https://github.com/lorg/sqladmindemo

Feel free to use it. Perhaps in the future we can incorporate it (or something like it) into sqladmin as a proper demo and capability show-case, plus it's a useful tool when developing sqladmin itself.

**Comment by GeorgiDimv** [comment]
Hey I tried to run it locally and after adding 2 dependencies to the requerment.txt file I got it working amazing.

Can you give me permission to create a branch and open PR in order for other ppl who might want to test it out.

From: Imri Goldberg ***@***.***>
Date: Tuesday, 29 April 2025 at 17:51
To: aminalaee/sqladmin ***@***.***>
Cc: Dimov, Georgi ***@***.***>, Mention ***@***.***>
Subject: Re: [aminalaee/sqladmin] column_filters (Discussion #374)

Hi @GeorgiDimv<https://github.com/GeorgiDimv> , I created a small demo project that I actually used for testing during development, I've now uploaded it here.
https://github.com/lorg/sqladmindemo

Feel free to use it. Perhaps in the future we can incorporate it (or something like it) into sqladmin as a proper demo and capability show-case, plus it's a useful tool when developing sqladmin itself.

**Reply by lorg** [reply]
I added you as a collaborator

**Comment by lorg** [comment]
@GeorgiDimv just letting you know that @aminalaee is back, I will implement the changes that he requested in the PR, and hopefully this will get merged in the next couple of days.

**Reply by ArfatAllarakha** [reply]
any update?

**Reply by lorg** [reply]
@ArfatAllarakha I completed the changes he requested, and now waiting for the PR to be reviewed again and hopefully merged. Thank you for reminding me!

**Comment by lorg** [comment]
Guys the PR has been merged. Thanks @aminalaee !!!
FYI @ArfatAllarakha @GeorgiDimv @gabriel-f-santos

**Reply by wnowicki** [reply]
looking forward to the release

**Reply by aminalaee** [reply]
Coming out in a few minutes

**Comment by chezou** [comment]
I created a follow up PR to introduce text input based column filter.
https://github.com/aminalaee/sqladmin/pull/945

**Reply by chezou** [reply]
@aminalaee If you have a chance, can you review the PR? I got a change request by random reviewer and I can see the reviewer creating change requests and approvals to many PRs, but I'm confusing to see it.

**Reply by chezou** [reply]
Thanks @aminalaee for merging my PR!

**Reply by chezou** [reply]
@aminalaee When you have a chance, it'd be great if you could release the new version so that we can reduce build time on CI 😄

**Reply by chezou** [reply]
Now 2.2.0 has been released! I believe this discussion can be closed now.
